source/geh_calculated_measurements/src/geh_calculated_measurements/net_consumption_group_6/domain/cnc_daily_calculation.py

```diff
@@ -0,0 +1,508 @@
+from datetime import datetime
+
+import pyspark.sql.functions as F
+import pyspark.sql.types as T
+from geh_common.domain.types import MeteringPointType
+from geh_common.pyspark.transformations import convert_from_utc, convert_to_utc
+from geh_common.telemetry import use_span
+from geh_common.testing.dataframes import testing
+from pyspark.sql import DataFrame, Window
+
+from geh_calculated_measurements.common.domain import ContractColumnNames, CurrentMeasurements
+from geh_calculated_measurements.common.domain.model import CalculatedMeasurementsDaily
+from geh_calculated_measurements.net_consumption_group_6.domain import (
+    ChildMeteringPoints,
+    ConsumptionMeteringPointPeriods,
+)
+
+
+@use_span()
+@testing()
+def execute(
+    consumption_metering_point_periods: ConsumptionMeteringPointPeriods,
+    child_metering_points: ChildMeteringPoints,
+    current_measurements: CurrentMeasurements,
+    time_zone: str,
+    execution_start_datetime: datetime,
+) -> CalculatedMeasurementsDaily:
+    # Filter and join the data
+    filtered_time_series = _filter_and_aggregate_daily(current_measurements)
+
+    parent_child_joined = _join_child_to_consumption(
+        consumption_metering_point_periods, child_metering_points, execution_start_datetime
+    )
+    filtered_time_series = convert_from_utc(filtered_time_series, time_zone)
+    parent_child_joined = convert_from_utc(parent_child_joined, time_zone)
+
+    # enforce 3 year
+    metering_points = _generate_metering_points_with_cut_off(parent_child_joined)
+
+    # split periods by settelement month
+    periods = _create_periods_for_settlement(metering_points)
```

Comment:
Why are all these comments needed? Is it because the function names ought to be improved?
I assume that you could save writing, reading and maintaing this comment by doing this:
```suggestion
    periods = _split_periods_by_settlement_month(metering_points)
```

Reply:
This comment applies to multiple of your comments. But surely not to all. Some are very useful and shouldn't be removed.